programs/vote/src/vote_state/mod.rs

```diff
@@ -382,6 +473,12 @@ impl VoteState {
             // to the current vote state root for safety.
             if earliest_slot_hash_in_history > new_proposed_root {
                 vote_state_update.root = self.root_slot;
+            } else if !slot_hashes
+                // Verify that the root is in slot hashes
+                .iter()
+                .any(|&(slot, _)| slot == new_proposed_root)
+            {
+                return Err(VoteError::RootOnDifferentFork);
```

Comment:
A couple things:

1) Can we add an equivalent test to `test_check_update_vote_state_slot_not_on_fork()` to test for this case

2) Because votes are executed so often, in the pursuit of perf, I would prefer not to iterate through all of the slot hashes again here. I think we can update the below loop to check for the root by mapping `vote_state_update_index = 0` to the root slot, and then updating all the checks that `vote_state_update_index` is involved in to avoid off by one errors

3) I think 1+2 are complicated enough to go into their own PR

Reply:
sounds good I'll add it to the upcoming pr